components/binding/internal/worker/targetkind.go

```diff
@@ -0,0 +1,64 @@
+package worker
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/kyma-project/kyma/components/binding/internal/storage"
+	"github.com/kyma-project/kyma/components/binding/pkg/apis/v1alpha1"
+	log "github.com/sirupsen/logrus"
+	"k8s.io/client-go/dynamic"
+)
+
+type TargetKindStorage interface {
+	Register(tk v1alpha1.TargetKind) error
+	Unregister(tk v1alpha1.TargetKind) error
+	Get(kind storage.Kind) (*storage.ResourceData, error)
+}
+
+type TargetKindWorker struct {
+	storage TargetKindStorage
+	dynamicClient dynamic.Interface
+}
+
+func NewTargetKindWorker(storage TargetKindStorage, dynamicClient dynamic.Interface) *TargetKindWorker {
+	return &TargetKindWorker{storage: storage,
+	dynamicClient: dynamicClient,
+	}
+}
+
+func (w *TargetKindWorker) Process(targetKind *v1alpha1.TargetKind, log log.FieldLogger) (*v1alpha1.TargetKind, error) {
+	log.Info("start TargetKind process")
+	// create client for kind and save to storage
+
+	if targetKind.Status.Registered {
+	_, err := w.storage.Get(storage.Kind(targetKind.Name))
```

Comment:
IMO is better to make a field `targetKind.Name` as `Kind` type inside `/pkg/apis/v1alpha1/targetkind_types.go` then you don't have to map all the time